So... who's going to open an upstream bug for this? =)

---
--- configure.ac
+++ configure.ac
@@ -20,33 +20,89 @@
 # GObject Introspection
 GOBJECT_INTROSPECTION_REQUIRE(0.6.5)
 
+dnl ----------
 dnl dbus
-PKG_CHECK_MODULES(DBUS, dbus-glib-1,
-		  have_dbus=true, have_dbus=false)
+dnl ----------
+AC_ARG_ENABLE(dbus,
+	      AS_HELP_STRING([--enable-dbus], [Build DBus gir data]))
+have_dbus=false
+if test "$enable_dbus" != "no"; then
+	PKG_CHECK_MODULES(DBUS, dbus-glib-1,
+			  have_dbus=true, have_dbus=false)
+
+	if test "$enable_dbus" = "yes" -a "$have_dbus" = "false"; then
+		AC_MSG_ERROR([DBus gir data requested but dbus-glib-1 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_DBUS, $have_dbus)
 
+dnl ----------
 dnl atk
-PKG_CHECK_MODULES(ATK, atk >= 1.12.0,
-                  have_atk=true, have_atk=false)
+dnl ----------
+AC_ARG_ENABLE(atk,
+	      AS_HELP_STRING([--enable-atk], [Build Atk gir data]))
+have_atk=false
+if test "$enable_atk" != "no"; then
+	PKG_CHECK_MODULES(ATK, atk >= 1.12.0,
+			  have_atk=true, have_atk=false)
+
+	if test "$enable_atk" = "yes" -a "$have_atk" = "false"; then
+		AC_MSG_ERROR([Atk gir data requested but atk not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_ATK, $have_atk)
 
+dnl ----------
 dnl pango
-PKG_CHECK_MODULES(PANGO, pango >= 1.16.0,
-                  have_pango=true, have_pango=false)
-AM_CONDITIONAL(BUILD_PANGO, $have_pango)
+dnl ----------
+AC_ARG_ENABLE(pango,
+	      AS_HELP_STRING([--enable-pango], [Build Pango gir data]))
+have_pango=false
+have_pangoxft=false
+if test "$enable_pango" != "no"; then
+	PKG_CHECK_MODULES(PANGO, pango >= 1.16.0,
+			  have_pango=true, have_pango=false)
+
+	if test "$enable_pango" = "yes" -a "$have_pango" = "false"; then
+		AC_MSG_ERROR([Pango gir data requested but pango not found])
+	fi
 
-PKG_CHECK_MODULES(PANGOXFT, pangoxft >= 1.16.0,
-		  have_pangoxft=true, have_pangoxft=false)
+	PKG_CHECK_MODULES(PANGOXFT, pangoxft >= 1.16.0,
+			  have_pangoxft=true, have_pangoxft=false)
+fi
+AM_CONDITIONAL(BUILD_PANGO, $have_pango)
 AM_CONDITIONAL(BUILD_PANGOXFT, $have_pangoxft)
 
+dnl ----------
 dnl poppler
-PKG_CHECK_MODULES(POPPLER, poppler-glib >= 0.8,
-                  have_poppler=true, have_poppler=false)
+dnl ----------
+AC_ARG_ENABLE(poppler,
+	      AS_HELP_STRING([--enable-poppler], [Build Poppler gir data]))
+have_poppler=false
+if test "$enable_poppler" != "no"; then
+	PKG_CHECK_MODULES(POPPLER, poppler-glib >= 0.8,
+			  have_poppler=true, have_poppler=false)
+
+	if test "$enable_poppler" = "yes" -a "$have_poppler" = "false"; then
+		AC_MSG_ERROR([Poppler gir data requested but poppler-glib not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_POPPLER, $have_poppler)
 
+dnl ----------
 dnl gtk+
-PKG_CHECK_MODULES(GTK, gtk+-2.0 >= 1.12.0,
-                  have_gtk=true, have_gtk=false)
+dnl ----------
+AC_ARG_ENABLE(gtk,
+	      AS_HELP_STRING([--enable-gtk], [Build GTK+ gir data]))
+have_gtk=false
+if test "$enable_gtk" != "no"; then
+	PKG_CHECK_MODULES(GTK, gtk+-2.0 >= 1.12.0,
+			  have_gtk=true, have_gtk=false)
+
+	if test "$enable_gtk" = "yes" -a "$have_gtk" = "false"; then
+		AC_MSG_ERROR([GTK+ gir data requested but gtk+-2.0 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_GTK, $have_gtk)
 GDK_TARGET=
 GDK_TARGET_GIRS=
@@ -61,49 +117,148 @@
 AC_SUBST(GDK_TARGET)
 AC_SUBST(GDK_TARGET_GIRS)
 
+dnl ----------
 dnl gconf
-PKG_CHECK_MODULES(GCONF, gconf-2.0,
-                  have_gconf=true, have_gconf=false)
+dnl ----------
+AC_ARG_ENABLE(gconf,
+	      AS_HELP_STRING([--enable-gconf], [Build GConf gir data]))
+have_gconf=false
+if test "$enable_gconf" != "no"; then
+	PKG_CHECK_MODULES(GCONF, gconf-2.0,
+			  have_gconf=true, have_gconf=false)
+
+	if test "$enable_gconf" = "yes" -a "$have_gconf" = "false"; then
+		AC_MSG_ERROR([GConf gir data requested but gconf-2.0 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_GCONF, $have_gconf)
 
+dnl ----------
 dnl soup
-PKG_CHECK_MODULES(SOUP, libsoup-2.4,
-                  have_soup=true, have_soup=false)
+dnl ----------
+AC_ARG_ENABLE(soup,
+	      AS_HELP_STRING([--enable-soup], [Build LibSoup gir data]))
+have_soup=false
+if test "$enable_soup" != "no"; then
+	PKG_CHECK_MODULES(SOUP, libsoup-2.4,
+			  have_soup=true, have_soup=false)
+
+	if test "$enable_soup" = "yes" -a "$have_soup" = "false"; then
+		AC_MSG_ERROR([LibSoup gir data requested but libsoup-2.4 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_SOUP, $have_soup)
 
+dnl ----------
 dnl babl
-PKG_CHECK_MODULES(BABL, babl,
-	 	  have_babl=true, have_babl=false)
+dnl ----------
+AC_ARG_ENABLE(babl,
+	      AS_HELP_STRING([--enable-babl], [Build Babl gir data]))
+have_babl=false
+if test "$enable_babl" != "no"; then
+	PKG_CHECK_MODULES(BABL, babl,
+			  have_babl=true, have_babl=false)
+
+	if test "$enable_babl" = "yes" -a "$have_babl" = "false"; then
+		AC_MSG_ERROR([Babl gir data requested but babl not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_BABL, $have_babl)
 
+dnl ----------
 dnl nautilus-extension
-PKG_CHECK_MODULES(NAUTILUS, libnautilus-extension,
-		  have_nautilusextension=true, have_nautilusextension=false)
+dnl ----------
+AC_ARG_ENABLE(nautilus,
+	      AS_HELP_STRING([--enable-nautilus], [Build Nautilus Extension gir data]))
+have_nautilusextension=false
+if test "$enable_nautilus" != "no"; then
+	PKG_CHECK_MODULES(NAUTILUS, libnautilus-extension,
+			  have_nautilusextension=true, have_nautilusextension=false)
+
+	if test "$enable_nautilus" = "yes" -a "$have_nautilusextension" = "false"; then
+		AC_MSG_ERROR([Nautilus Extension gir data requested but libnautilus-extension not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_NAUTILUS, $have_nautilusextension)
 
+dnl ----------
 dnl gnome-keyring
-PKG_CHECK_MODULES(GNOMEKEYRING, gnome-keyring-1,
-                  have_gnomekeyring=true, have_gnomekeyring=false)
+dnl ----------
+AC_ARG_ENABLE(gnomekeyring,
+	      AS_HELP_STRING([--enable-gnomekeyring], [Build gnome-keyring gir data]))
+have_gnomekeyring=false
+if test "$enable_gnomekeyring" != "no"; then
+	PKG_CHECK_MODULES(GNOMEKEYRING, gnome-keyring-1,
+			  have_gnomekeyring=true, have_gnomekeyring=false)
+
+	if test "$enable_gnomekeyring" = "yes" -a "$have_gnomekeyring" = "false"; then
+		AC_MSG_ERROR([gnome-keyring gir data requested but gnome-keyring-1 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_GNOMEKEYRING, $have_gnomekeyring)
 
+dnl ----------
 dnl webkit
-PKG_CHECK_MODULES(WEBKIT, webkit-1.0 >= 1.0,
-                  have_webkit=true, have_webkit=false)
+dnl ----------
+AC_ARG_ENABLE(webkit,
+	      AS_HELP_STRING([--enable-webkit], [Build Webkit-Gtk gir data]))
+have_webkit=false
+if test "$enable_webkit" != "no"; then
+	PKG_CHECK_MODULES(WEBKIT, webkit-1.0 >= 1.0,
+			  have_webkit=true, have_webkit=false)
+
+	if test "$enable_webkit" = "yes" -a "$have_webkit" = "false"; then
+		AC_MSG_ERROR([WebKit-Gtk gir data requested but webkit-1.0 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_WEBKIT, $have_webkit)
 
+dnl ----------
 dnl notify
-PKG_CHECK_MODULES(NOTIFY, libnotify,
-                  have_notify=true, have_notify=false)
+dnl ----------
+AC_ARG_ENABLE(notify,
+	      AS_HELP_STRING([--enable-libnotify], [Build LibNotify gir data]))
+have_notify=false
+if test "$enable_notify" != "no"; then
+	PKG_CHECK_MODULES(NOTIFY, libnotify,
+			  have_notify=true, have_notify=false)
+
+	if test "$enable_notify" = "yes" -a "$have_notify" = "false"; then
+		AC_MSG_ERROR([LibNotify gir data requested but libnotify not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_NOTIFY, $have_notify)
 
+dnl ----------
 dnl gnio
-PKG_CHECK_MODULES(GNIO, gnio,
-                  have_gnio=true, have_gnio=false)
+dnl ----------
+AC_ARG_ENABLE(gnio,
+	      AS_HELP_STRING([--enable-gnio], [Build gnio gir data]))
+have_gnio=false
+if test "$enable_gnio" != "no"; then
+	PKG_CHECK_MODULES(GNIO, gnio,
+			  have_gnio=true, have_gnio=false)
+
+	if test "$enable_gnio" = "yes" -a "$have_gnio" = "false"; then
+		AC_MSG_ERROR([gnio gir data requested but gnio not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_GNIO, $have_gnio)
 
+dnl ----------
 dnl clutter
-PKG_CHECK_MODULES(CLUTTER, clutter-0.8 >= 0.8,
-                  have_clutter=true, have_clutter=false)
+dnl ----------
+AC_ARG_ENABLE(clutter,
+	      AS_HELP_STRING([--enable-clutter], [Build Clutter gir data]))
+have_clutter=false
+if test "$enable_clutter" != "no"; then
+	PKG_CHECK_MODULES(CLUTTER, clutter-0.8 >= 0.8,
+			  have_clutter=true, have_clutter=false)
+
+	if test "$enable_clutter" = "yes" -a "$have_clutter" = "false"; then
+		AC_MSG_ERROR([Clutter gir data requested but clutter-0.8 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_CLUTTER, $have_clutter)
 CLUTTER_BACKEND=
 CLUTTER_BACKEND_GIRS=
@@ -126,73 +281,160 @@
 AC_SUBST(CLUTTER_BACKEND)
 AC_SUBST(CLUTTER_BACKEND_GIRS)
 
+dnl ----------
 dnl clutter-gtk
-PKG_CHECK_MODULES(CLUTTERGTK, clutter-gtk-0.8 >= 0.8,
-                  have_clutter_gtk=true, have_clutter_gtk=false)
+dnl ----------
+AC_ARG_ENABLE(clutter_gtk,
+	      AS_HELP_STRING([--enable-clutter-gtk], [Build Clutter-Gtk gir data]))
+have_clutter_gtk=false
+if test "$enable_clutter_gtk" != "no"; then
+	PKG_CHECK_MODULES(CLUTTERGTK, clutter-gtk-0.8 >= 0.8,
+			  have_clutter_gtk=true, have_clutter_gtk=false)
+
+	if test "$enable_clutter_gtk" = "yes" -a "$have_clutter_gtk" = "false"; then
+		AC_MSG_ERROR([Clutter-Gtk gir data requested but clutter-gtk-0.8 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_CLUTTERGTK, $have_clutter_gtk)
 
+dnl ----------
 dnl clutter-cairo
-PKG_CHECK_MODULES(CLUTTERCAIRO, clutter-cairo-0.8 >= 0.8,
-                  have_clutter_cairo=true, have_clutter_cairo=false)
+dnl ----------
+AC_ARG_ENABLE(clutter_cairo,
+	      AS_HELP_STRING([--enable-clutter-cairo], [Build Clutter-Cairo gir data]))
+have_clutter_cairo=false
+if test "$enable_clutter_cairo" != "no"; then
+	PKG_CHECK_MODULES(CLUTTERCAIRO, clutter-cairo-0.8 >= 0.8,
+			  have_clutter_cairo=true, have_clutter_cairo=false)
+
+	if test "$enable_clutter_cairo" = "yes" -a "$have_clutter_cairo" = "false"; then
+		AC_MSG_ERROR([Clutter-Cairo gir data requested but clutter-cairo-0.8 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_CLUTTERCAIRO, $have_clutter_cairo)
 
+dnl ----------
 dnl gstreamer
-PKG_CHECK_MODULES(GSTREAMER, gstreamer-0.10 >= 0.10.0, 
-                  have_gstreamer=true, have_gstreamer=false)
-AM_CONDITIONAL(BUILD_GSTREAMER, $have_gstreamer)
-
-dnl gstbufferlist.h, gsttaskpool.h were added in 0.10.24 (and in
-dnl prereleases of that)
+dnl ----------
+AC_ARG_ENABLE(gstreamer,
+	      AS_HELP_STRING([--enable-gstreamer], [Build Gstreamer gir data]))
+have_gstreamer=false
+have_gstreamer_plugins_base=false
 have_gstbufferlist_h=false
-if $have_gstreamer ; then
-   save_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS $GSTREAMER_CFLAGS"
-   AC_CHECK_HEADER(gst/gstbufferlist.h, [have_gstbufferlist_h=true])
-   CPPFLAGS="$save_CPPFLAGS"
-fi
-AM_CONDITIONAL(HAVE_GSTBUFFERLIST_H, $have_gstbufferlist_h)
-
 have_gsttaskpool_h=false
-if $have_gstreamer ; then
-   save_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS $GSTREAMER_CFLAGS"
-   AC_CHECK_HEADER(gst/gsttaskpool.h, [have_gsttaskpool_h=true])
-   CPPFLAGS="$save_CPPFLAGS"
+if test "$enable_gstreamer" != "no"; then
+	PKG_CHECK_MODULES(GSTREAMER, gstreamer-0.10 >= 0.10.0, 
+			  have_gstreamer=true, have_gstreamer=false)
+
+	if test "$enable_gstreamer" = "yes" -a "$have_gstreamer" = "false"; then
+		AC_MSG_ERROR([Gstreamer gir data requested but gstreamer-0.10 not found])
+	fi
+
+	dnl gstbufferlist.h, gsttaskpool.h were added in 0.10.24 (and in
+	dnl prereleases of that)
+	if $have_gstreamer ; then
+		save_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS $GSTREAMER_CFLAGS"
+		AC_CHECK_HEADER(gst/gstbufferlist.h, [have_gstbufferlist_h=true])
+		CPPFLAGS="$save_CPPFLAGS"
+	fi
+
+	if $have_gstreamer ; then
+		save_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS $GSTREAMER_CFLAGS"
+		AC_CHECK_HEADER(gst/gsttaskpool.h, [have_gsttaskpool_h=true])
+		CPPFLAGS="$save_CPPFLAGS"
+	fi
+
+	PKG_CHECK_MODULES(GSTREAMER_PLUGINS_BASE, gstreamer-plugins-base-0.10 >= 0.10.0, 
+			  have_gstreamer_plugins_base=true, 
+			  have_gstreamer_plugins_base=false)
 fi
+AM_CONDITIONAL(HAVE_GSTBUFFERLIST_H, $have_gstbufferlist_h)
 AM_CONDITIONAL(HAVE_GSTTASKPOOL_H, $have_gsttaskpool_h)
-
-dnl gstreamer
-PKG_CHECK_MODULES(GSTREAMER_PLUGINS_BASE, gstreamer-plugins-base-0.10 >= 0.10.0, 
-                  have_gstreamer_plugins_base=true, 
-                  have_gstreamer_plugins_base=false)
+AM_CONDITIONAL(BUILD_GSTREAMER, $have_gstreamer)
 AM_CONDITIONAL(BUILD_GSTREAMER_PLUGINS_BASE, $have_gstreamer_plugins_base)
 
+dnl ----------
 dnl gtksourceview
-PKG_CHECK_MODULES(GTKSOURCEVIEW, gtksourceview-2.0,
-                  have_gtksourceview=true, have_gtksourceview=false)
+dnl ----------
+AC_ARG_ENABLE(gtksourceview,
+	      AS_HELP_STRING([--enable-gtksourceview], [Build gtksourceview gir data]))
+have_gtksourceview=false
+if test "$enable_gtksourceview" != "no"; then
+	PKG_CHECK_MODULES(GTKSOURCEVIEW, gtksourceview-2.0,
+			  have_gtksourceview=true, have_gtksourceview=false)
+
+	if test "$enable_gtksourceview" = "yes" -a "$have_gtksourceview" = "false"; then
+		AC_MSG_ERROR([gtksourceview gir data requested but gtksourceview-2.0 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_GTKSOURCEVIEW, $have_gtksourceview)
 
+dnl ----------
 dnl vte
-PKG_CHECK_MODULES(VTE, vte,
-                  have_vte=true, have_vte=false)
+dnl ----------
+AC_ARG_ENABLE(vte,
+	      AS_HELP_STRING([--enable-vte], [Build Vte gir data]))
+have_vte=false
+if test "$enable_vte" != "no"; then
+	PKG_CHECK_MODULES(VTE, vte,
+			  have_vte=true, have_vte=false)
+
+	if test "$enable_vte" = "yes" -a "$have_vte" = "false"; then
+		AC_MSG_ERROR([Vte gir data requested but vte not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_VTE, $have_vte)
 
+dnl ----------
 dnl goocanvas
-PKG_CHECK_MODULES(GOOCANVAS, goocanvas,
-                  have_goocanvas=true, have_goocanvas=false)
+dnl ----------
+AC_ARG_ENABLE(goocanvas,
+	      AS_HELP_STRING([--enable-goocanvas], [Build goocanvas gir data]))
+have_goocanvas=false
+if test "$enable_goocanvas" != "no"; then
+	PKG_CHECK_MODULES(GOOCANVAS, goocanvas,
+			  have_goocanvas=true, have_goocanvas=false)
+
+	if test "$enable_goocanvas" = "yes" -a "$have_goocanvas" = "false"; then
+		AC_MSG_ERROR([goocanvas gir data requested but goocanvas not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_GOOCANVAS, $have_goocanvas)
 
+dnl ----------
 dnl mission-control (disabled for now, it has odd structs and isn't useful yet)
-dnl PKG_CHECK_MODULES(MISSIONCONTROL, libmissioncontrol,
-dnl                  have_missioncontrol=true, have_missioncontrol=false)
+dnl ----------
+dnl AC_ARG_ENABLE(missioncontrol,
+dnl 	      AS_HELP_STRING([--enable-missioncontrol], [Build missioncontrol gir data]))
+dnl have_missioncontrol=false
+dnl if test "$enable_missioncontrol" != "no"; then
+dnl 	PKG_CHECK_MODULES(MISSIONCONTROL, libmissioncontrol,
+dnl	                  have_missioncontrol=true, have_missioncontrol=false)
+dnl
+dnl	if test "$enable_missioncontrol" = "yes" -a "$have_missioncontrol" = "false"; then
+dnl		AC_MSG_ERROR([missioncontrol gir data requested but libmissioncontrol not found])
+dnl	fi
+dnl fi
 have_missioncontrol=false
 AM_CONDITIONAL(BUILD_MISSIONCONTROL, $have_missioncontrol)
 
-dnl gssdp
-PKG_CHECK_MODULES(GSSDP, gssdp-1.0,
-                  have_gssdp=true, have_gssdp=false)
+dnl ----------
+dnl gssdp & gupnp
+dnl ----------
+AC_ARG_ENABLE(gssdp,
+	      AS_HELP_STRING([--enable-gssdp], [Build gssdp & gupnp gir data]))
+have_gssdp=false
+if test "$enable_gssdp" != "no"; then
+	PKG_CHECK_MODULES(GSSDP, gssdp-1.0,
+			  have_gssdp=true, have_gssdp=false)
+
+	if test "$enable_gssdp" = "yes" -a "$have_gssdp" = "false"; then
+		AC_MSG_ERROR([gssdp gir data requested but gssdp-1.0 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_GSSDP, $have_gssdp)
 
-dnl gupnp
-if test $have_gssdp; then
+if test "$have_gssdp" = "true"; then
     PKG_CHECK_MODULES(GUPNP, gupnp-1.0,
                       have_gupnp=true, have_gupnp=false)
 else
@@ -200,24 +442,68 @@
 fi
 AM_CONDITIONAL(BUILD_GUPNP, $have_gupnp)
 
+dnl ----------
 dnl avahi-gobject
-PKG_CHECK_MODULES(AVAHI, avahi-core >= 0.6 avahi-gobject >= 0.6,
-                  have_avahi=true, have_avahi=false)
+dnl ----------
+AC_ARG_ENABLE(avahi,
+	      AS_HELP_STRING([--enable-avahi], [Build avahi gir data]))
+have_avahi=false
+if test "$enable_avahi" != "no"; then
+	PKG_CHECK_MODULES(AVAHI, avahi-core >= 0.6 avahi-gobject >= 0.6,
+			  have_avahi=true, have_avahi=false)
+
+	if test "$enable_avahi" = "yes" -a "$have_avahi" = "false"; then
+		AC_MSG_ERROR([Avahi gir data requested but avahi-core and avahi-gobject not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_AVAHI, $have_avahi)
 
+dnl ----------
 dnl unique
-PKG_CHECK_MODULES(UNIQUE, unique-1.0 >= 1.0.0,
-                  have_unique=true, have_unique=false)
+dnl ----------
+AC_ARG_ENABLE(unique,
+	      AS_HELP_STRING([--enable-unique], [Build LibUnique gir data]))
+have_unique=false
+if test "$enable_unique" != "no"; then
+	PKG_CHECK_MODULES(UNIQUE, unique-1.0 >= 1.0.0,
+			  have_unique=true, have_unique=false)
+
+	if test "$enable_unique" = "yes" -a "$have_unique" = "false"; then
+		AC_MSG_ERROR([LibUnique gir data requested but unique-1.0 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_UNIQUE, $have_unique)
 
+dnl ----------
 dnl gmenu
-PKG_CHECK_MODULES(GMENU, libgnome-menu,
-		  have_gmenu=true, have_gmenu=false)
+dnl ----------
+AC_ARG_ENABLE(gmenu,
+	      AS_HELP_STRING([--enable-gmenu], [Build libgnome-menu gir data]))
+have_gmenu=false
+if test "$enable_gmenu" != "no"; then
+	PKG_CHECK_MODULES(GMENU, libgnome-menu,
+			  have_gmenu=true, have_gmenu=false)
+
+	if test "$enable_gmenu" = "yes" -a "$have_gmenu" = "false"; then
+		AC_MSG_ERROR([libgnome-menu gir data requested but libgnome-menu not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_GMENU, $have_gmenu)
 
+dnl ----------
 dnl wnck
-PKG_CHECK_MODULES(WNCK, libwnck-1.0,
-		  have_wnck=true, have_wnck=false)
+dnl ----------
+AC_ARG_ENABLE(wnck,
+	      AS_HELP_STRING([--enable-wnck], [Build libwnck gir data]))
+have_wnck=false
+if test "$enable_wnck" != "no"; then
+	PKG_CHECK_MODULES(WNCK, libwnck-1.0,
+			  have_wnck=true, have_wnck=false)
+
+	if test "$enable_wnck" = "yes" -a "$have_wnck" = "false"; then
+		AC_MSG_ERROR([libwnck gir data requested but libwnck-1.0 not found])
+	fi
+fi
 AM_CONDITIONAL(BUILD_WNCK, $have_wnck)
 
 AC_CONFIG_FILES([Makefile
